src/Analyser/NodeScopeResolver.php

```diff
@@ -925,7 +930,17 @@ private function processStmtNode(
 			$classStatementsGatherer = new ClassStatementsGatherer($classReflection, $nodeCallback);
 			$this->processAttributeGroups($stmt, $stmt->attrGroups, $classScope, $classStatementsGatherer);
 
-			$this->processStmtNodes($stmt, $stmt->stmts, $classScope, $classStatementsGatherer, $context);
+			// analyze static methods first, constructor next and instance methods last so we can carry over the scope
+			$classLikeStatements = $stmt->stmts;
+			usort($classLikeStatements, static function ($a, $b) {
+				if (!$a instanceof Node\Stmt\ClassMethod || !$b instanceof Node\Stmt\ClassMethod) {
+					return 0;
+				}
+
+				return [!$a->isStatic(), $a->name->toLowerString() !== '__construct'] <=> [!$b->isStatic(), $b->name->toLowerString() !== '__construct'];
```

Comment:
We have a better way to ask for constructor, that works for same-named constructors as class on PHP 7.

Reply:
I looked into adding `ExtendedMethodReflection->isConstructor(): TrinaryLogic`, but this is incompatible with 
[`PhpMethodFromParserNodeReflection->isConstructor(): bool`
](https://github.com/phpstan/phpstan-src/blob/da74773df7198bb61a69b65af26dd68186b1f91b/src/Reflection/Php/PhpMethodFromParserNodeReflection.php#L279).

any preferences how you want this to be resolved?

Reply:
I think we can leave as it is. This is mostly useful for readonly properties anyway, which means `__construct()` constructors...